### PR TITLE
*: support rust-protobuf

### LINKED ISSUE #385: Support rust-protobuf
### Feature Request

Support `rust-protobuf` for `protobuf` codec, to integrate `client-rust` with other projects using `rust-protobuf`.

For example, integrate with https://github.com/tikv/tikv for integration tests of full transaction features.

Also refer to https://github.com/tikv/client-rust/issues/122.

### PR DESCRIPTION
**Pull request by pingyu**
Close #385 

### Changes:
- Add feature `prost-codec` & `protobuf-codec` to select codec.
- Make code compatible with both `prost-codec` & `protobuf-codec`.
- Add tests for `protobuf-codec` in CI.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,69 +10,37 @@ jobs:
   check:
     name: check
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        env: [ '', 'RUST_PROTOBUF=1' ]
```
Comment (iosmanthus) [inline .github/workflows/ci.yml:15]:
```suggestion
        env: [ 'RUST_PROTOBUF=0', 'RUST_PROTOBUF=1' ]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,69 +10,37 @@ jobs:
...
+        run: ${{ matrix.env }} make check
 
   unit-test:
     name: unit test
     env:
       CARGO_INCREMENTAL: 0
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        env: [ '', 'RUST_PROTOBUF=1' ]
```
Comment (iosmanthus) [inline .github/workflows/ci.yml:43]: Ditto

**Review by iosmanthus — COMMENTED** [review]
Rest LGTM

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pingyu) [reply .github/workflows/ci.yml]: Fixed. PTAL~

**Review by iosmanthus — APPROVED** [review]

**Pull request merged**